### PR TITLE
[SPARK-25656][SQL][DOC][EXAMPLE][BRANCH-2.4] Add a doc and examples about extra data source options

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
## What changes were proposed in this pull request?

Our current doc does not explain how we are passing the data source specific options to the underlying data source. According to [the review comment](https://github.com/apache/spark/pull/22622#discussion_r222911529), this PR aims to add more detailed information and examples. This is a backport of #22801. `orc.column.encoding.direct` is removed since it's not supported in ORC 1.5.2.

## How was this patch tested?

Manual.

### CODE REVIEW

File: examples/src/main/r/RSparkSQLExample.R [hunk 1]
```diff
@@ -114,10 +114,14 @@ write.df(namesAndAges, "namesAndAges.parquet", "parquet")
...
 # $example on:manual_load_options_csv$
-df <- read.df("examples/src/main/resources/people.csv", "csv", sep=";", inferSchema=T, header=T)
+df <- read.df("examples/src/main/resources/people.csv", "csv", sep = ";", inferSchema = TRUE, header = TRUE)
 namesAndAges <- select(df, "name", "age")
 # $example off:manual_load_options_csv$
 
+# $example on:manual_save_options_orc$
+df <- read.df("examples/src/main/resources/users.orc", "orc")
+write.orc(df, "users_with_options.orc", orc.bloom.filter.columns = "favorite_color", orc.dictionary.key.threshold = 1.0)
+# $example off:manual_save_options_orc$
```
Comment (dongjoon-hyun) [inline examples/src/main/r/RSparkSQLExample.R]: Hi, @felixcheung . This is a backport of #22801 (on master branch).
Could you review this?

File: docs/sql-data-sources-load-save-functions.md [hunk 2]
```diff
@@ -82,6 +82,49 @@ To load a CSV file you can use:
...
+CREATE TABLE users_with_options (
+  name STRING,
+  favorite_color STRING,
+  favorite_numbers array<integer>
+) USING ORC
+OPTIONS (
+  orc.bloom.filter.columns 'favorite_color',
+  orc.dictionary.key.threshold '1.0'
+)
+{% endhighlight %}
```
Comment (dongjoon-hyun) [inline docs/sql-data-sources-load-save-functions.md]: Hi, @dbtsai . This is a backport of #22801 without `orc.column.encoding.direct`.

**Comment by dbtsai** [comment]
Thanks @dongjoon-hyun This LGTM!

**Comment by dongjoon-hyun** [comment]
Thank you so much for review, @dbtsai .
Merged to `branch-2.4`.

**Pull request closed without merging**